content/blog/2023/2023-08-30-fairness-equal-odds-postprocessing.md

```diff
@@ -0,0 +1,57 @@
+longExcerpt = "Explore the Equal Odds Postprocessing widget in Orange, designed to fine-tune your model's fairness. We explain how the algorithm operates and showcase its effectiveness with an example using the German credit dataset."
++++
+
+
+In the [previous blog post]({{< ref "2023-08-28-fairness-adversarial-debiasing.md" >}}), we discussed the Adversarial Debiasing model, a bias-aware model. This blog post will discuss the Equal Odds Postprocessing widget, a bias-aware post-processor, which can be used with any model to mitigate bias in its predictions.
+
+### Equal Odds Postprocessing:
+
+The [Equal Odds Postprocessing](https://arxiv.org/abs/1610.02413) widget is a post-processing type of fairness mitigation algorithm designed to ensure fairness in supervised learning. It modifies the predictions of any given classifier to meet certain fairness criteria, specifically focusing on "Equalized Odds" or more relaxed criteria like Equal Opportunity. Because it is a post-processing algorithm, it does not require any support from the model (like reweighing, which requires the model to support weights) or changes to the model (like adversarial debiasing, which requires the model to be a neural network). This makes it a versatile algorithm that can be used with any model.
+
+The widget Equalized Odds Postprocessing represents a novelty in the Orange environment, as until now, there has been no widget with similar functionality. Its operation is similar to other Orange widgets representing models, except it also expects one of the Orange models as an input. It operates by first learning from the predictions of the model's training data. Then, it applies the Equalized Odds Postprocessing algorithm from the [AIF360](https://aif360.res.ibm.com/) library to the predictions the selected model makes on the test data. This ensures that the adjusted predictions conform to the fairness definition of equalized odds.
+
+In more detail, the Equalized Odds Postprocessing algorithm works by solving a linear program with some constraints and the following objective function:
+
+`c = [fpr_0 - tpr_0, tnr_0 - fnr_0, fpr_1 - tpr_1, tnr_1 - fnr_1]`
+
+Where `fpr`, `tpr`, `tnr`, and `fnr` are the false positive, true positive, true negative, and false negative rates for privileged (`0`) and unprivileged (`1`) groups.
+
+The algorithm then finds the optimal solution to the linear program, which results in a set of probabilities with which to flip the model's predictions to equalize the odds of being correctly or incorrectly classified for both privileged and unprivileged groups:
+
+- `sp2p`: Probability of flipping a label from positive to negative for the privileged group.
```

Comment:
Probability of flipping a label is repeating. Reformulate to avoid such repetitions.